### ISSUE TITLE
discussion: Require explicit version references in the registry for GitHub modules

### ISSUE THREAD

**Issue by lucacasonato** [denoland/dotland#250]
I propose that we require a `@[version]` tag for `/x` and `/std` urls. This would mean we remove the automatic aliasing from `https://deno.land/x/std/manual.md` to `https://deno.land/x/std@master/manual.md` for registry modules with the type `github`. If a user tries to request a non explicit version reference the CF Worker should respond with a 400 error and the text `You should specify an explicit version reference when requesting a module (e.g. https://deno.land/x/std@v0.34.0/readme.md)`.

`https://deno.land/x/std@master/manual.md` would still be valid for importing from master, you just have to be explicit.

On the website when a user accesses `https://deno.land/x/std/manual.md` we should automatically and explicitly redirect them to the latest version (we get this from the GitHub API on the client). We should also add a dropdown to the top of the registry page that the user can use to switch between all of the available git tags and master.

The background of this change is to not have users implicitly use the `master` version of a module, but rather explicitly choose what version they want to use. What is the point of creating versions if everyone just uses `master` at all times anyway? This automatic master aliasing has caused issues before: https://github.com/denoland/deno/issues/4089, https://github.com/denoland/deno/issues/3987.

/cc @nayeemrmn @ry @bartlomieju

**Comment by jimsimon** [comment]
I agree with this proposed change. I think master should be removed entirely from the drop-down of options and it should instead default to the most recent versions/tag/branch.

Having this default to master is opening up authors (including the authors of the standard libs) to a lot of potential pain. If/When a breaking change is published to master, people are going to file issues and complain they their code no longer works.  Even if it's technically the fault of the person using the library, it's going to be perceived as instability.  Additionally, repeatedly having to respond to these issues will be a drain on development resources.

Personal experience: I actually didn't even notice the version drop-down the first time I started pulling package URLs from the STD lib docs. One of the libs was erroring out all over the place and my first impression was "wow, I guess this isn't quite ready for use after all". Fortunately I decided to double check some things and realized I wasn't using a released version, but I suspect others may not.
